externalpolicy/text_language/text_language.go

```diff
@@ -85,287 +82,189 @@ func (p *TextConstraintLanguagePlugin) Validate(dconstraints interface{}) (bool,
-		if operator := tokenContainsOperator(token); operator != "" {
-			newToken := strings.Trim(strings.Replace(token, operator, fmt.Sprintf(" %v ", operator), -1), " ")
-			s = strings.Replace(s, token, newToken, -1)
+	if lexMap["OpIn"] == opType {
+		if lexMap["ListStr"] != valType && lexMap["QuoteStr"] != valType && lexMap["VersRange"] != valType && lexMap["Vers"] != valType {
+			return fmt.Errorf("The 'in' operator can only be used for types version and list of string")
+		}
+		if lexMap["VersRange"] == valType {
+			// Using the factory function to validate version ranges
+			_, err = semanticversion.Version_Expression_Factory(val.(string))
+			if err != nil {
+				return err
+			}
 		}
 	}
 
-	return s
+	return nil
 }
 
-// 1. == is supported for all types except list of string, which would use 'in'.
-// 2. for numeric types, the operators ==, <, >, <=, >= are supported
-// 3. false and true are the only valid values for a boolean type
-// 4. for string types, a quoted string, inside which is a list of comma separated strings provide acceptable values
-// 5. string values that contain spaces must be quoted
-// 6. for the version type, supported values are a single version or a range of versions in the semantic version format (the same as used for service verions). The == operator implies that the value is a single version. The 'in' operator treats the value as a version range. As with service versions, the version 1.0.0 when treated as a version range is equivalent to the explicit range [1.0.0,INFINITY).
+func getLexer() lexer.Definition {
+	return lexer.Must(ebnf.New(`
+	  alphanumeric = digit | alpha .
 
-func validateOneConstraintExpression(expression string) (bool, error) {
-	if len(expression) == 0 {
-		return true, nil
-	}
+	  vers = {digit} "." {digit} "." {digit} .
+	  digit = "0"…"9" .
+	  alpha = "a"…"z" | "A"…"Z" .
 
-	// get message printer because this function is called by CLI
-	msgPrinter := i18n.GetMessagePrinter()
-
-	pieces := strings.Split(expression, " ")
-	if len(pieces) < 3 {
-		return false, errors.New(msgPrinter.Sprintf("found %v token(s), expecting 3 in an expression %v, expected form is expression with 3 tokens: <property> <operator> <value> in constraint expression", len(pieces), expression))
-	}
-
-	compOp := pieces[1]
-	value := pieces[2]
+	  AndOp = whitespace ("AND" | "&&") whitespace .
+	  OrOp = whitespace ("OR" | "||") whitespace .
 
-	// if will failed on case when string values that contain spaces but not quoted (starting from 2nd interation)
-	if !isAllowedComparisonOpType(pieces[1]) {
-		return false, errors.New(msgPrinter.Sprintf("Expression: %v should contain valid comparison operator - wrong operator %v. Allowed operators: %v, %v, %v, %v, %v, %v, %v, %v", expression, pieces[1], lessthan, greaterthan, equalto, doubleequalto, lessthaneq, greaterthaneq, notequalto, inoperator))
-	}
+	  OpEq =  [whitespace]  ( "="["="] | "!=" )   .
+	  OpComp =  [whitespace] ("<"["="] | ">"["="]) [whitespace] .
```

Comment:
Tweak this to support either >= or =>, same for less than.

Reply:
Done